datalad/support/annexrepo.py

```diff
@@ -180,31 +184,64 @@ def __init__(self, path, url=None, runner=None,
           Short description that humans can use to identify the
           repository/location, e.g. "Precious data on my laptop"
         """
-        # initialize
-        self._uuid = None
-        self._annex_common_options = []
-
-        if annex_opts or annex_init_opts:
-            lgr.warning("TODO: options passed to git-annex and/or "
-                        "git-annex-init are currently ignored.\n"
-                        "options received:\n"
-                        "git-annex: %s\ngit-annex-init: %s" %
-                        (annex_opts, annex_init_opts))
 
+        # BEGIN Repo validity test
+        # We want to fail early for tests, that would be performed a lot. In particular this is about
+        # AnnexRepo.is_valid_repo. We would use the latter to decide whether or not to call AnnexRepo() only for
+        # __init__ to then test the same things again. If we fail early we can save the additional test from outer
+        # scope.
         fix_it = False
+        do_init = False
         try:
             super(AnnexRepo, self).__init__(
                 path, url, runner=runner,
                 create=create, create_sanity_checks=create_sanity_checks,
                 repo=repo, git_opts=git_opts, fake_dates=fake_dates)
         except GitCommandError as e:
+            # XXX Ben: This seems outdated, since cloning shouldn't happen in GitRepo() anymore.
+            #          Needs double-check whether this is true and whether it needs to be moved to the clone routine.
```

Comment:
It would be great to get this removed, if that is confirmed.

Reply:
I'm on it. Moved error handling to clone routine. Running tests locally ...